app/src/main/java/com/sunasterisk/fooddaily/ui/adapter/FoodAdapter.kt

```diff
@@ -0,0 +1,65 @@
+package com.sunasterisk.fooddaily.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sunasterisk.fooddaily.R
+import com.sunasterisk.fooddaily.data.model.FoodDetail
+import com.sunasterisk.fooddaily.utils.Constants
+
+class FoodAdapter(
+    private val foodList: List<FoodDetail>,
+    private val viewType: Int
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    class DailyMenuViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        fun bind(food: FoodDetail) {
+            //TODO bind data from foodList to DailyMenu
+        }
+    }
+
+    class OtherFoodListViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        fun bind(food: FoodDetail) {
+            //TODO bind data from foodList to OtherFoodList
+        }
+    }
+
+    class OtherFoodGridViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        fun bind(food: FoodDetail) {
+            //TODO bind data from foodList to OtherFoodGrid
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int = viewType
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            Constants.FOOD_TYPE_DAILY_MENU -> {
+                DailyMenuViewHolder(
+                    LayoutInflater.from(parent.context)
+                        .inflate(R.layout.item_daily_menu_food, parent, false)
+                )
+            }
+            Constants.FOOD_TYPE_OTHER_LIST -> {
+                OtherFoodListViewHolder(
+                    LayoutInflater.from(parent.context)
+                        .inflate(R.layout.item_other_food_list, parent, false)
+                )
+            }
+            else -> OtherFoodGridViewHolder(
+                LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_other_food_grid, parent, false)
+            )
+        }
+    }
+
+    override fun getItemCount(): Int = foodList.size
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is DailyMenuViewHolder -> holder.bind(foodList[position])
+            is OtherFoodListViewHolder -> holder.bind(foodList[position])
+            is OtherFoodGridViewHolder -> holder.bind(foodList[position])
+        }
+    }
```

Comment:
```suggestion
    override fun onBindViewHolder(holder: FoodAdapter.FoodViewHolder, position: Int) {
        holder.bind(foodList[position])
    }
```

Tạo 1 base class cho 3 thằng ViewHolder kia:
```kotlin
abstract class FoodViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
    abstract fun bind(food: FoodDetail)
}
```